### PR TITLE
Add plotting with charming.

### PR DESCRIPTION
**Pull request by lambertsbennett**
# Description
Add initial plotting command using charming.  Right now it works only with filesize value types. Need to extend it. But you can try it out with some cool outcomes.  ```let a = ls; | plot rose $a.name $a.size ```

image

# User-Facing Changes
Add plot rose command.

### CODE REVIEW

**Comment by lambertsbennett** [comment]
I really like Charming and I think that the possibilities are really vast. There needs to be some work done both on the Charming side and on the Nushell side to get this working really nicely.

**Comment by lambertsbennett** [comment]
Made some small changes, now you can also do things like ```let a = (ps | sort-by cpu | last 5); | plot rose $a.name $a.cpu ```.

**Comment by fdncred** [comment]
This is so super fun to play with and that rose chart is beautiful. Love the interactivity too, which adds a lot to it. Keep up the good work.

File: crates/nu-command/src/plot/plot_rose.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                "labels",
+                SyntaxShape::String,
+                "category field for the rose plot",
+            )
+            .required("values", SyntaxShape::Any, "value field for the rose plot")
+    }
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "creates a rose plot from Nu data",
+            example: "let a = ls; | plot rose $a.type $a.size",
```
Comment (amtoine) [inline crates/nu-command/src/plot/plot_rose.rs:34]: early feeback, but this is a bit confusing i must admit :open_mouth: 
i would write either
```nushell
let a = ls; plot rose $a.type $a.size
```
or
```nushell
ls | plot rose $in.type $in.size
```
:yum:

Comment (lambertsbennett) [reply crates/nu-command/src/plot/plot_rose.rs:34]: Ah I didn't know you could do the second one! I'll change that!

**Comment by sholderbach** [comment]
Looking at the volume of additional dependencies we would need to accept into the core nushell for an isolated set of commands, leads me to favor this being a plugin focussed on plotting/visualizations for now.

There the additional compile time/binary size etc. for users that don't regularly use it wouldn't be a concern. Also the development could iterate a bit faster without the stricter quality standards we have to apply for the monolithic `nu` binary.

In the future we could certainly revisit such a decision around inclusion when the dependency tree is pruned and the users are eager to have that included after enjoying it as a plugin.

**Comment by lambertsbennett** [comment]
@sholderbach ok I'll move it then into its own repo and make a plugin!

**Comment by amtoine** [comment]
thanks @lambertsbennett for understanding :pray: 
can't wait to play with `plot ...` as a plugin :partying_face: 

i'll close this then :relieved:

**Comment by fdncred** [comment]
@lambertsbennett Please post the url to the repo you're going to use. I'd love to watch is and test whatever you create. I'm a big fan of charting.

**Comment by lambertsbennett** [comment]
@fdncred I'll work on it here: https://github.com/lambertsbennett/nu_plugin_charming. Probably get started tomorrow!

**Comment by fdncred** [comment]
ok, thanks for listing it. I'll be watching. Can't wait to play more with it.

**Comment by fdncred** [comment]
@lambertsbennett I stepped away for a moment, how's the charting plugin coming?

**Comment by lambertsbennett** [comment]
@fdncred hey sorry I haven't made much progress in the last weeks. Had to focus on some other stuff. My hope is to have a basically working plugin by Christmas.

**Comment by fdncred** [comment]
Sounds good. No pressure. I just really liked the direction you were going. I'll try to be patient. 🤣

**Pull request closed without merging**